pallets/parachain-staking/src/lib.rs

```diff
@@ -880,8 +885,9 @@ pub mod pallet {
 				})
 				.collect();
 			ensure!(in_top, Error::<T>::DelegationDNE);
+			top_delegations.total = top_delegations.total.saturating_add(more);
```

Comment:
This looks to be the place where we fix the bug. What does the previous code do then?

Reply:
Yes it is, the previous code does NOT update the total top delegations. The new code DOES update the total top delegations.

Whether or not top delegations total is updated is what bleeds into the collator.total_counted and leads to the bug.